src/Scanner.hpp

```diff
@@ -22,7 +23,29 @@ struct ScanInput
 struct ScanOutput
 {
     int m_IncludedFileCount;
+
+    //The memory management guarantee here is that the array this points to came from a scratch allocator, so it will dissapear on you,
+    //but the actual string payloads the const char*'s are pointing at are guaranteed to stay alive until the end of the build.
     const FileAndHash *m_IncludedFiles;
 };
 
-bool ScanImplicitDeps(StatCache *stat_cache, const ScanInput *input, ScanOutput *output);
+// This callback will be called for any newly discovered (ie, not loaded from cache)
+// includes found during scanning.
+// `includingFile` is the file which contains the include statement
+// `includedFile` is the file being included.
+// The callback returns a bool, which tells the scanner if it should ignore this included file.
+// If the callback returns true, the included file will be ignored, which means it will not be added
```

Comment:
s/ignored/filtered/ ?

Reply:
I think "ignored" is more explicit really. Filtered could be read as "filtered out" or "filtered in".